### DISCUSSION TITLE
Is there support for ADS1115 ADC 4 Channels Module ADC Analog to Digital

### DISCUSSION THREAD

**Discussion by jpikoulas** [meshtastic/Discussions#144] [General]
Hi , is there any support or plans to add support for the [ADS1115](https://www.amazon.co.uk/eMagTech-Converter-Development-Compatible-Raspberry/dp/B0D1721D6H/ref=sr_1_10?crid=212CWCERRGNON&dib=eyJ2IjoiMSJ9.VlTC0q_5mC6sU-F_HrstDtc6XPRtGjKnnAf1Aacbe_Y1WO69Nrn5WwRiHGmvNKbX0Nn2t1bBMbEiJqxUhFOLjpCq2lzUPalID1X6uBL4SSXFsmBpiWa-hRTgQy_tGafyBcfUyZOg1vBF2k7GJSd5AdUYHBxKi7HDDWlSoeqVaD0xQ3NOdDRvwbWAhbWyTl9rH50Vss7CG6gDPBamkFZhwWJc0sd4yB3OHiDzHSXK_ZnnSXIMghegNlx3FCl9EDrUxFjf8UaoJ78UWewuvyPotciBXxG68zf0UENd98gOqa1tegMK597wZz3ogoiJe_7Yz1-EAp8vt9MVWggM-fa3NLtXKGNgbWJnm9ngPH6N5rxgGLmMpg0ZS7amc4IaCsCJ-i6OGKopxqf4U59INBlabiRmKdFH8uBtjPNEDEnI8GKjOOIrl9UztDPgvuexRWu8.ziWeVWvfgoDDhbKySeKl-TvW2JuS3YPU8lKE8TCN208&dib_tag=se&keywords=ads1115&nsdOptOutParam=true&qid=1734865050&sprefix=ads1115%2Caps%2C275&sr=8-10) ADC 16 Bit 4 Channels Module ADC Analog to Digital Converter Development Board .
I have searched on different forums could not find anything, basically was looking for a way to read analog soil sensors and pass this as telemetry metrics over LoRa.
Any ideas/suggestions will be much appreciated.

**Comment by garthvh** [comment]
Sensors need to be I2c so they can be auto detected and provide instant read data that is more conducive to sending over the mesh. If you have an alalog sensor you want to use the serial module and a seperate mcu to do the processing is the way to go.

**Reply by jpikoulas** [reply]
This is a I2C sensor,  [ADS1115 Module, 16-bit ADC, I2C Interface, 4 Channels](https://www.amazon.com/ADS1115-Module-16-bit-Interface-Channels/dp/B0D8BF6KRF)

**Reply by garthvh** [reply]
> This is a I2C sensor, [ADS1115 Module, 16-bit ADC, I2C Interface, 4 Channels](https://www.amazon.com/ADS1115-Module-16-bit-Interface-Channels/dp/B0D8BF6KRF)

Yes but it is passing raw unprocessed analog signals via I2c, this data is unlikely to be able to be sent over a low bandwidth mesh without additional processing. This would not allow any analog sensor to be added, would still require a library or the additional processing would need to be done on an external MCU. We are already low on space for additional sensor types.

**Reply by jpikoulas** [reply]
OK, kind of make sense. Is  it a way to use the ESP32S3 ADC readings, instead of the ADS1115 module?
